### PR TITLE
skip merging of parent-child spans if span.kind is producer or consumer

### PR DESCRIPTION
**Pull request by ashishagg**

### CODE REVIEW

File: reader/src/test/scala/com/expedia/www/haystack/trace/reader/unit/readers/transformers/ServerClientSpanMergeTransformerSpec.scala [hunk 1]
```diff
@@ -195,8 +231,23 @@ class ServerClientSpanMergeTransformerSpec extends BaseUnitTestSpec with ValidTr
       spanTree.children.head.children.map(_.span) should contain allOf(underlyingSpans.apply(2), underlyingSpans.apply(3))
       spanTree.children.head.children.foreach(tree => tree.children.size shouldBe 0)
     }
+
+    it ("should not merge producer and consumer parent-child spans") {
```
Comment (adriancole) [inline reader/src/test/scala/com/expedia/www/haystack/trace/reader/unit/readers/transformers/ServerClientSpanMergeTransformerSpec.scala]: I guess before it was merging regardless of span ID? or were you getting producer and consumer with the same ID?

Comment (ashishagg) [reply reader/src/test/scala/com/expedia/www/haystack/trace/reader/unit/readers/transformers/ServerClientSpanMergeTransformerSpec.scala]: We need a fix for a usecase where a singled produced message in kafka gets consumed by multiple consumer apps. The merge logic in haystack works good for a client-server span but for this kafka usecase, we can't merge the spans as there is no one consumer.

Comment (adriancole) [reply reader/src/test/scala/com/expedia/www/haystack/trace/reader/unit/readers/transformers/ServerClientSpanMergeTransformerSpec.scala]: right I mean to say usually they have different span IDs so I guess in your scenario multiple consumers are still using the same span ID (hence being merged) right?

Comment (ashishagg) [reply reader/src/test/scala/com/expedia/www/haystack/trace/reader/unit/readers/transformers/ServerClientSpanMergeTransformerSpec.scala]: right.

**Pull request merged**